R/data-combined.R

```diff
@@ -0,0 +1,676 @@
+#' that the datasets whose names you wish to not change should be specified as
+#' `NULL` in the list.
+#'
+#'
+#' @import tidyr
+#' @import ospsuite.utils
+#'
+#' @examples
+#'
+#' # load the simulation
+#' simFilePath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simFilePath)
+#' simulationResults <- runSimulations(simulations = sim)
+#'
+#' # create a new dataset object
+#' dataSet <- DataSet$new()
+#'
+#' # created object with datasets combined
+#' myCombDat <- DataCombined$new()
+#' myCombDat$addSimulationResults(simulationResults)
+#' myCombDat$addDataSets(dataSet)
+#'
+#' # print the object
+#' myCombDat
+#' @docType class
+#' @export
+
+DataCombined <- R6::R6Class(
+  classname = "DataCombined",
+  inherit = Printable,
+
+  # public fields and methods ------------------------------------
+
+  public = list(
+
+    ## setter methods ---------------
+
+    #' @param dataSets Instance (or a `list` of instances) of the `DataSet`
+    #'   class.
+    #'
+    #' @description
+    #' Adds observed data.
+    #'
+    #' @return `DataCombined` object containing observed data.
+    addDataSets = function(dataSets, names = NULL, groups = NULL) {
+      # if list of instances is provided, `purrr::walk()` will check each element
+      # `purrr::walk()` will in no way modify objects provided to it
+      if (is.list(dataSets)) {
+        purrr::walk(.x = dataSets, .f = ~ validateIsOfType(.x, "DataSet", FALSE))
+      } else {
+        validateIsOfType(dataSets, "DataSet", FALSE)
+      }
+
+      # validate vector arguments' type and length
+      groups <- validateVectorArgs(groups, objCount(dataSets), type = "character")
+      names <- validateVectorArgs(names, objCount(dataSets), type = "character")
+
+      # if alternate names are provided for datasets, use them instead
+      #
+      # for `NULL` elements in a list, the original dataset names will be used
+      #
+      # to get these names, go inside each element of the list and extract
+      # (`purrr::pluck()`) the name from the object itself and use them instead
+      # and simplify to a character vector (using `purrr::map_chr()`)
+      if (!is.null(names) && is.list(dataSets)) {
+        names <- ifelse(is.na(names), purrr::map_chr(dataSets, ~ purrr::pluck(.x, "name")), names)
+      }
+
+      # Update private fields for the new setter call
+
+      private$.dataCombined <- private$.updateDF(private$.dataCombined, private$.dataSetToDF(dataSets, names, groups))
+      private$.dataCombined <- private$.extractXYData(private$.dataCombined)
+      private$.groupMap <- private$.extractGroupMap(private$.dataCombined)
+      private$.names <- private$.extractNames(private$.dataCombined)
+
+      # set up data transformations
+      self$setDataTransformations(names)
+
+      # for method chaining
+      invisible(self)
+    },
+
+    #' @param simulationResults Object of type `SimulationResults` produced by
```

Comment:
Can't you use @inherit to inherit the documentation from `getOutputValues`? This would ensure that the documentation is always up to date and save some lines of code.

Reply:
I had tried this. Unfortunately, this is not supported for `R6`. 

In general, `{roxygen2}` doesn't have a good support for R6,as can be surmised from [dozens of pending issues](https://github.com/r-lib/roxygen2/issues?q=is%3Aopen+is%3Aissue+label%3A%22R6+%3Asix%3A%22).

Reply:
I've added `TODO` marker here to revisit this in case this is supported by _roxygen2_.